### PR TITLE
Prototype (Character) Classes

### PR DESCRIPTION
**Pull request by ThoughtbotThom**
Constructiong D&D character classes through prototypes.

### CODE REVIEW

**Comment by fbm-static** [comment]
This repo is drifting away from TIL to write-as-much-tutorials-as-possible

**Comment by ThoughtbotThom** [comment]
I'm sorry you feel that way, however, this is something I recently learned and the practical context in which it was employed.

File: javascript/prototypingCharacterClasses.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#Prototype Your (Character) Class#
+
+Continuing in using Javascript to build D&D characters, I decided to use prototypes.
```
Comment (thorncp) [inline javascript/prototypingCharacterClasses.md]: Reading this in isolation, I don't know what you're continuing from. Is this part of a series or referring to a previous TIL?

Comment (ThoughtbotThom) [inline javascript/prototypingCharacterClasses.md]: [This](https://github.com/thoughtbot/til/blob/master/javascript/funWithFunctions.md); not really a series; beyond that it's related n that I wrote it, is D&D themed and Javascript based.

Comment (JoelQ) [inline javascript/prototypingCharacterClasses.md]: Thoughts on linking to the previous TIL?

Comment (ThoughtbotThom) [inline javascript/prototypingCharacterClasses.md]: Not at all opposed!

File: javascript/prototypingCharacterClasses.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Character.prototype = {
+	level: function (){ 
+	this.lvl++; 
+	this.hp = this.hd + this.mod[2] + ((this.lvl-1)*(((this.hd/2)+1) + this.mod[2])) + (this.dr * this.lvl); 
+	if ((this.lvl - 1) % 4 == 0 ) {
+		this.profBonus++;
+		this.spellSave = 8 + this.profBonus + this.mod[this.castingAbility];
+		this.spellAttack = this.profBonus + this.mod[this.castingAbility];
+	}
+},
```
Comment (JoelQ) [inline javascript/prototypingCharacterClasses.md]: Indentation is off for the body of this function

File: javascript/prototypingCharacterClasses.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	this.spellSave = 8 + this.profBonus + this.mod[castingAbility];
+	this.spellAttack = this.profBonus + this.mod[castingAbility];
+}
+``` 
+	
+In using a prototype over an object we gain the ability to efficiently store identical functions common to all characters without bloating objects.
+
+```javascript
+Character.prototype = {
+	level: function (){
```
Comment (JoelQ) [inline javascript/prototypingCharacterClasses.md]: Thoughts on renaming this function `levelUp` to make it obvious this is changing the character, not just retrieving the level?

File: javascript/prototypingCharacterClasses.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+}
+``` 
+	
+In using a prototype over an object we gain the ability to efficiently store identical functions common to all characters without bloating objects.
+
+```javascript
+Character.prototype = {
+	level: function (){ 
+	this.lvl++; 
+	this.hp = this.hd + this.mod[2] + ((this.lvl-1)*(((this.hd/2)+1) + this.mod[2])) + (this.dr * this.lvl);
```
Comment (JoelQ) [inline javascript/prototypingCharacterClasses.md]: Thoughts on extracting some local variables here? This formula is quite complex and would be easier to understand if it had named parts to it.

**Comment by JoelQ** [comment]
Looks good to me. Squash, merge and buffer at your leisure.

**Pull request merged**